api/reporter/reporter.go

```diff
@@ -0,0 +1,89 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package reporter
+
+import (
+	"fmt"
+	"github.com/submariner-io/submariner-operator/internal/cli"
+)
+
+// Reporter is responsible for reporting back on the progress of the specified preparation.
+type Reporter interface {
+	// Started will report that an operation started.
+	Started(message string, args ...interface{})
```

Comment:
Can we keep these as imperatives? `Start()`, `Succeed()` etc.

Reply:
Also the word _Queued_  seems tied to the existing CLI spinner implementation. I suggest simplifying to:

```
Start(message string, args ...interface{})
Success(message string, args ...interface{})
Failure(message string, args ...interface{})
Warning(message string, args ...interface{})
End()
```